New "?const time_t *? {aka ?const long long int *?} but argument is of type ?long int" issues with compiling the VE7FET AX25 suite on Raspberry Pi OS Trixie 32bit Lite (No GUI)

I received two new compiling error reports for the VE7FET AX.25 repo using a fully updated 32bit Lite Raspberry Pi OS for ARM.  The compile errors in ax25apps and ax25tools are easily reproduced:

--David


ax25apps:
--------------------------------------------------------------------------------------------------------------------------

---removing generated files---
---autoreconf---
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: /usr/bin/autoconf --force
configure.ac:15: warning: The macro 'AC_PROG_GCC_TRADITIONAL' is obsolete.
configure.ac:15: You should run autoupdate.
./lib/autoconf/c.m4:1676: AC_PROG_GCC_TRADITIONAL is expanded from...
configure.ac:15: the top level
autoreconf: running: /usr/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:14: installing './compile'
configure.ac:39: installing './config.guess'
configure.ac:39: installing './config.sub'
configure.ac:5: installing './install-sh'
configure.ac:5: installing './missing'
ax25ipd/Makefile.am: installing './depcomp'
autoreconf: Leaving directory '.'
---removing generated files---
---autoreconf---
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: /usr/bin/autoconf --force
configure.ac:15: warning: The macro 'AC_PROG_GCC_TRADITIONAL' is obsolete.
configure.ac:15: You should run autoupdate.
./lib/autoconf/c.m4:1676: AC_PROG_GCC_TRADITIONAL is expanded from...
configure.ac:15: the top level
autoreconf: running: /usr/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:14: installing './compile'
configure.ac:39: installing './config.guess'
configure.ac:39: installing './config.sub'
configure.ac:5: installing './install-sh'
configure.ac:5: installing './missing'
ax25ipd/Makefile.am: installing './depcomp'
autoreconf: Leaving directory '.'
Compiling on 4 CPUs concurrently

 dpkg-buildpackage -us -uc -ui -j4
dpkg-buildpackage: info: source package ax25apps
dpkg-buildpackage: info: source version 2.0.1
dpkg-buildpackage: info: source distribution unstable
dpkg-buildpackage: info: source changed by Lee Woldanski <ve7fet@tparc.org>
 dpkg-source --before-build .
dpkg-buildpackage: info: host architecture armhf
 debian/rules clean
dh clean --with autoreconf
   dh_clean
 dpkg-source -b .
dpkg-source: info: using source format '3.0 (native)'
dpkg-source: info: building ax25apps in ax25apps_2.0.1.tar.xz
dpkg-source: info: building ax25apps in ax25apps_2.0.1.dsc
 debian/rules binary
dh binary --with autoreconf
   dh_update_autotools_config
   dh_autoreconf
configure.ac:15: warning: The macro 'AC_PROG_GCC_TRADITIONAL' is obsolete.
configure.ac:15: You should run autoupdate.
./lib/autoconf/c.m4:1676: AC_PROG_GCC_TRADITIONAL is expanded from...
configure.ac:15: the top level
   dh_auto_configure
        ./configure --build=arm-linux-gnueabihf --prefix=/usr --includedir=\${prefix}/include --mandir=\${prefix}/share/man --infodir=\${prefix}/share/info --sysconfdir=/etc --localstatedir=/var --disable-option-checking --disable-silent-rules --libdir=\${prefix}/lib/arm-linux-gnueabihf --libexecdir=\${prefix}/lib/arm-linux-gnueabihf --disable-maintainer-mode --disable-dependency-tracking
checking for a BSD-compatible install... /usr/bin/install -c
checking whether sleep supports fractional seconds... yes
checking filesystem timestamp resolution... 0.01
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking xargs -n works... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... none
checking for ax25_config_load_ports in -lax25... yes
checking for initscr in -lncursesw... yes
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking for sys/wait.h that is POSIX.1 compatible... yes
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for vfork.h... no
checking for sys/time.h... yes
checking for sys/select.h... yes
checking for sys/socket.h... yes
checking for utime.h... yes
./configure: line 5010: #include: command not found
checking for arpa/inet.h... yes
checking for fcntl.h... yes
checking for limits.h... yes
checking for memory.h... yes
checking for netdb.h... yes
checking for netinet/in.h... yes
checking for sys/ioctl.h... yes
checking for sys/socket.h... (cached) yes
checking for sys/time.h... (cached) yes
checking for syslog.h... yes
checking for termio.h... yes
checking for termios.h... yes
checking for utime.h... (cached) yes
checking for ncursesw/curses.h... yes
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking for off_t... yes
checking whether struct tm is in sys/time.h or time.h... time.h
checking for working volatile... yes
checking for pid_t... yes
checking for fork... yes
checking for vfork... yes
checking for alarm... yes
checking for working fork... yes
checking for working vfork... (cached) yes
checking build system type... arm-unknown-linux-gnueabihf
checking host system type... arm-unknown-linux-gnueabihf
checking for GNU libc compatible malloc... yes
checking for working memcmp... yes
checking for working mktime... yes
checking types of arguments for select... int,fd_set *,struct timeval *
checking whether lstat correctly handles trailing slash... yes
checking whether stat accepts an empty string... no
checking whether utime accepts a null argument... yes
checking for bzero... yes
checking for gethostbyname... yes
checking for inet_ntoa... yes
checking for memchr... yes
checking for memmove... yes
checking for memset... yes
checking for select... yes
checking for socket... yes
checking for strcasecmp... yes
checking for strchr... yes
checking for strdup... yes
checking for strerror... yes
checking for strrchr... yes
checking for strtol... yes
checking for strtoul... yes
checking for utime... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating call/Makefile
config.status: creating ax25ipd/Makefile
config.status: creating listen/Makefile
config.status: creating Makefile
config.status: creating ax25rtd/Makefile
config.status: creating ax25mond/Makefile
config.status: creating doc/Makefile
config.status: creating man/Makefile
config.status: creating etc/Makefile
config.status: creating ax25apps.spec
config.status: creating config.h
config.status: executing depfiles commands
   dh_auto_build
        make -j4
make[1]: Entering directory '/usr/src/archive/Rpi-scratch/linuxax25/ax25apps'
make  all-recursive
.
.
.
gcc -DHAVE_CONFIG_H -I. -I..   -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/usr/src/archive/Rpi-scratch/linuxax25/ax25apps=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -Wall -pedantic -c -o rosedump.o rosedump.c
gcc -DHAVE_CONFIG_H -I. -I..   -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/usr/src/archive/Rpi-scratch/linuxax25/ax25apps=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -Wall -pedantic -c -o flexnetdump.o flexnetdump.c
gcc -DHAVE_CONFIG_H -I. -I..   -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/usr/src/archive/Rpi-scratch/linuxax25/ax25apps=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -Wall -pedantic -c -o opentracdump.o opentracdump.c
gcc -DHAVE_CONFIG_H -I. -I..   -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/usr/src/archive/Rpi-scratch/linuxax25/ax25apps=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -Wall -pedantic -c -o utils.o utils.c
opentracdump.c: In function ?decode_timestamp?:
opentracdump.c:154:47: error: passing argument 1 of ?ctime? from incompatible pointer type [-Wincompatible-pointer-types]
  154 |         lprintf(T_OPENTRAC, "Time: %s", ctime(&rawtime));
      |                                               ^~~~~~~~
      |                                               |
      |                                               long int *
In file included from /usr/include/features.h:523,
                 from /usr/include/arm-linux-gnueabihf/bits/libc-header-start.h:33,
                 from /usr/include/stdio.h:28,
                 from opentracdump.c:7:
/usr/include/time.h:186:14: note: expected ?const time_t *? {aka ?const long long int *?} but argument is of type ?long int *?
  186 | extern char *__REDIRECT_NTH (ctime, (const time_t *__timer), __ctime64);
      |              ^~~~~~~~~~~~~~
make[3]: *** [Makefile:411: opentracdump.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/usr/src/archive/Rpi-scratch/linuxax25/ax25apps/listen'
make[2]: *** [Makefile:380: all-recursive] Error 1
make[2]: Leaving directory '/usr/src/archive/Rpi-scratch/linuxax25/ax25apps'
make[1]: *** [Makefile:319: all] Error 2
make[1]: Leaving directory '/usr/src/archive/Rpi-scratch/linuxax25/ax25apps'
dh_auto_build: error: make -j4 returned exit code 2
make: *** [debian/rules:16: binary] Error 25



ax25tools:
--------------------------------------------------------------------------------------------------------------------------
gcc -DHAVE_CONFIG_H -I. -I..  -D_GNU_SOURCE -DAX25_SYSCONFDIR=\""/etc/ax25/"\" -DAX25_LOCALSTATEDIR=\""/var/ax25/"\"  -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -Wdate-t
ime -D_FORTIFY_SOURCE=2  -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/usr/src/archive/Rpi-scratch/linuxax25/ax25tools=. -fstack-protector-strong -fstack-clash-protect
ion -Wformat -Werror=format-security -Wall -pedantic -c -o kissparms.o kissparms.c
gcc -DHAVE_CONFIG_H -I. -I..  -D_GNU_SOURCE -DAX25_SYSCONFDIR=\""/etc/ax25/"\" -DAX25_LOCALSTATEDIR=\""/var/ax25/"\"  -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -Wdate-t
ime -D_FORTIFY_SOURCE=2  -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/usr/src/archive/Rpi-scratch/linuxax25/ax25tools=. -fstack-protector-strong -fstack-clash-protect
ion -Wformat -Werror=format-security -Wall -pedantic -c -o mkiss.o mkiss.c
mkiss.c: In function ?report?:
mkiss.c:386:14: error: passing argument 1 of ?time? from incompatible pointer type [-Wincompatible-pointer-types]
  386 |         time(&t);
      |              ^~
      |              |
      |              long int *
In file included from /usr/include/features.h:523,
                 from /usr/include/arm-linux-gnueabihf/bits/libc-header-start.h:33,
                 from /usr/include/stdio.h:28,
                 from mkiss.c:40:
/usr/include/time.h:85:15: note: expected ?time_t *? {aka ?long long int *?} but argument is of type ?long int *?
   85 | extern time_t __REDIRECT_NTH (time, (time_t *__timer), __time64);
      |               ^~~~~~~~~~~~~~
mkiss.c:388:55: error: passing argument 1 of ?ctime? from incompatible pointer type [-Wincompatible-pointer-types]
  388 |         syslog(LOG_INFO, "Status report at %s", ctime(&t));
      |                                                       ^~
      |                                                       |
      |                                                       long int *
/usr/include/time.h:186:14: note: expected ?const time_t *? {aka ?const long long int *?} but argument is of type ?long int *?
  186 | extern char *__REDIRECT_NTH (ctime, (const time_t *__timer), __ctime64);
      |              ^~~~~~~~~~~~~~
mkiss.c: In function ?main?:
mkiss.c:675:68: warning: ISO C does not support the ?%m? gnu_printf format [-Wformat=]
  675 |                                         syslog(LOG_ERR, "tty->fd: %m");
      |                                                                    ^
mkiss.c:702:80: warning: ISO C does not support the ?%m? gnu_printf format [-Wformat=]
  702 |                                                 syslog(LOG_ERR, "pty[%d]->fd: %m\n", i);
      |                                                                                ^
make[3]: *** [Makefile:428: mkiss.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/usr/src/archive/Rpi-scratch/linuxax25/ax25tools/kiss'
make[2]: *** [Makefile:396: all-recursive] Error 1
make[2]: Leaving directory '/usr/src/archive/Rpi-scratch/linuxax25/ax25tools'
make[1]: *** [Makefile:335: all] Error 2
make[1]: Leaving directory '/usr/src/archive/Rpi-scratch/linuxax25/ax25tools'
dh_auto_build: error: make -j4 returned exit code 2

--- a/ax25apps/listen/opentracdump.c
+++ b/ax25apps/listen/opentracdump.c
@@ -148,7 +148,7 @@ static int decode_position(unsigned char *element, int element_len)
 static int decode_timestamp(unsigned char *element, int element_len)
 {
 	/* 0x11 Timestamp - Unix format time (unsigned)  */
-	long rawtime = 0;
+	time_t rawtime = 0;
 
 	rawtime = get32(element);
 	lprintf(T_OPENTRAC, "Time: %s", ctime(&rawtime));

--- a/ax25tools/6pack/m6pack.c
+++ b/ax25tools/6pack/m6pack.c
@@ -383,7 +383,7 @@ static void sigusr1_handler(int sig)
 static void report(struct iface *tty, struct iface **pty, int numptys)
 {
 	int i;
-	long t;
+	time_t t;
 
 	time(&t);
 	syslog(LOG_INFO, "version %s.", VERSION);

--- a/ax25tools/kiss/mkiss.c
+++ b/ax25tools/kiss/mkiss.c
@@ -381,7 +381,7 @@ static void sigusr1_handler(int sig)
 static void report(void)
 {
 	int i;
-	long t;
+	time_t t;
 
 	time(&t);
 	syslog(LOG_INFO, "version %s.", VERSION);

